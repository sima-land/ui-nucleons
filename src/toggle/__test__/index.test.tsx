import { fireEvent, render } from '@testing-library/react';
import { createRef } from 'react';
import { Toggle } from '..';

describe('<Toggle />', () => {
  it('should work without props', () => {
    const { container } = render(<Toggle />);

    expect(container).toMatchSnapshot();
  });

  it('should handle "defaultChecked" prop', () => {
    const { container } = render(<Toggle defaultChecked />);

    expect(container.querySelector('input')?.defaultChecked).toBe(true);
  });

  it('should handle "checked" prop', () => {
    const spy = jest.fn();
    const { container, rerender } = render(<Toggle checked onChange={spy} />);

    expect(container.querySelector('input')?.checked).toBe(true);

    rerender(<Toggle checked={false} onChange={spy} />);

    expect(container.querySelector('input')?.checked).toBe(false);
  });

  it('should handle "onChange" prop', () => {
    const spy = jest.fn();
    const { container } = render(<Toggle onChange={spy} />);

    expect(spy).toHaveBeenCalledTimes(0);

    fireEvent.click(container.querySelector('input') as HTMLElement);

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should handle "ref" prop', () => {
    const ref = createRef<HTMLInputElement>();

    const { container } = render(<Toggle ref={ref} />);

    expect(ref.current).toBe(container.querySelector('input'));
  });
});
