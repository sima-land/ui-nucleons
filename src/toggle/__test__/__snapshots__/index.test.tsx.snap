// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should work without props 1`] = `
<div>
  <label
    class="root"
  >
    <input
      class="input"
      data-testid="toggle"
      type="checkbox"
    />
    <span
      class="switch"
    />
  </label>
</div>
`;
